### ISSUE TITLE
CBF Controller for Double-Integrator dynamics

### ISSUE THREAD

**Issue by luigiberducci** [MIT-REALM/neural_clbf#33]
Hi all :-)
I have been implementing a simple training script for neural cbf controllers on double-integrator dynamics.
Since this is a quite standard example in most safety-critical control paper, I think it is of general interest.

I struggle in obtaining a good CBF candidate.
As this is the first use of this repository, any feedback on the implementation is welcome.

### Changes
- Implemented a new system with double-integrator dynamics.
- Created script for training a neural cbf controller.
- To avoid using gurobi, the `disable_gurobi` argument is added to  the CBFController and then forwarded to the parent class CLFController.

### Experiments
I run the script as it is and collected snapshot of the tensorboard after 50 epochs.

Despite the overall trend seems to go in the right direction and the positive and negative regions in the barrier function roughly correspond to safe/unsafe set, I cannot see sign of life in the rollouts. 
Do you have any insights on this?

**Accuracy**
![image](https://github.com/user-attachments/assets/29cf660a-7dae-436a-bf66-89dc4e6e8219)

**QP-relaxation**
![image](https://github.com/user-attachments/assets/6ab9c624-a173-40fc-9b1f-ea8e14032920)

**Loss**
![image](https://github.com/user-attachments/assets/7d5144dc-381b-47b8-bff7-a7ea2c8e49c6)

**Rollouts**
![image](https://github.com/user-attachments/assets/ac7ba8d0-47ce-4077-bb37-808513bc7792)

**V-contour:**
![image](https://github.com/user-attachments/assets/36ee097a-3dc4-48ba-bde6-6c81678e4365)
